scripts/metrics/pytest_metrics_xml_to_csv.py

```diff
@@ -0,0 +1,173 @@
+import sys
+import csv
+import argparse
+import xml.etree.ElementTree as xmlElementTree
+import datetime
+
+import ly_test_tools.cli.codeowners_hint as codeowners_hint
+from common import logging, exception
+
+
+TESTING_DIR = 'Testing'
+
+
+def _get_default_csv_filename():
+    """
+    Returns a default filename to be saved in the format of YYYY_MM_DD_HH_mm_pytest.csv
+    :return: The filename as a string
+    """
+    # Format default file name based off of date
+    now = datetime.datetime.now(tz=datetime.timezone.utc)
+    return f"{now.year}_{now.month:02d}_{now.day:02d}_{now.hour:02d}_{now.minute:02d}_pytest.csv"
+
+
+# Setup logging.
+logger = logging.get_logger("test_metrics")
+logging.setup_logger(logger)
+
+# Create the csv field header
+PYTEST_FIELDS_HEADER = [
+    'test_name',
+    'status',
+    'duration_seconds',
+    'sig_owner'
+]
+SIG_OWNER_CACHE = {}
+
+
+def main():
+    """
+    Creates the folder structure for metrics to be saved to S3 and converts Pytest xml's into csv format. This script
+    does not handle uploading of metrics.
+
+    :return: None
+    """
+    # Parse args
+    args = parse_args()
+
+    if not os.path.exists(args.xml_folder):
+        raise exception.MetricsExn(f"Cannot find directory: {args.xml_folder}")
+
+    # Create csv file
+    full_path = os.path.join(args.output_directory, args.branch)
+    if args.weeks:
+        week = datetime.datetime.now().isocalendar().week
+        full_path = os.path.join(full_path, f"Week{week:02d}")
+    full_path = os.path.join(full_path, args.csv_file)
+    if os.path.exists(full_path):
+        logger.warning(f"The file {full_path} already exists. It will be overridden.")
+    if not os.path.exists(os.path.dirname(full_path)):
+        # Create directory if it doesn't exist
+        os.makedirs(os.path.dirname(full_path))
+
+    # Create csv file
+    if os.path.exists(args.csv_file):
+        logger.warning(f"The file {args.csv_file} already exists. It will be overriden.")
+    with open(full_path, 'w', encoding='UTF8', newline='') as csv_file:
+        writer = csv.DictWriter(csv_file, fieldnames=PYTEST_FIELDS_HEADER, restval='N/A')
+        writer.writeheader()
+
+        # Parse Pytest xml's and write to csv file
+        for filename in os.listdir(args.xml_folder):
+            if filename.endswith('.xml'):
+                parse_pytest_xmls_to_csv(os.path.join(args.xml_folder, filename), writer)
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        description='This script assumes that Pytest xml files have been produced. The xml files will be parsed and '
+                    'written to a csv file.')
+    parser.add_argument(
+        'xml_folder',
+        help="Path to where the Pytest xml files live. O3DE CTest defaults this to {build_folder}/Testing/Pytest"
+    )
+    parser.add_argument(
+        "--csv-file", action="store", default=_get_default_csv_filename(),
+        help=f"The file name for the csv to be saved."
```

Comment:
Why is it important to allow the user to override the filename? Is the timestamp not critical to the metrics schema? If it is important to provide the option, this should probably suggest why yet recommend relying on the default.

Reply:
Having an override is convenient for local testing, I can add a recommendation for default in the comments